### ISSUE TITLE
[BUG]: [Rechecker] Detection bypass by Hooked/Injected client.dll in Half-Life 1

### ISSUE THREAD

**Issue by abdobiskraKGB** [s1lentq/rechecker#4]
### 🤔 What happened?

Hello,

I am experiencing a detection bypass in Rechecker v2.7 on ReHLDS (Build 3.14.0.857-dev).

The Core Issue:
The cheat being used features a Hooked/Injected client.dll. This modified DLL is not being caught by standard MD5 hashing in resources.ini.

Technical Observation:
When I use the path cl_dlls/client.dll with an UNKNOWN flag to log hashes, the server either:

    Receives a Spoofed Hash (the cheat sends a hash of a clean file to trick the server).

    Fails to respond to the query entirely, even though the file is active in the game process.

Since this client.dll is integrated with an external cheat executable.

Evidence & Resources:

    Video Demonstration (Cheat in action): https://www.youtube.com/watch?v=AvuWa_gi2NI

    Observed Behavior: Legitimate players return correct hashes (e.g., 311ed0af), but this specific "Hooked" client bypasses the check or remains invisible to the Rechecker logic.

Request:
Is there a way to implement a Memory-based check or a deeper consistency check through Rechecker that can detect if the client.dll has been tampered with in the game's memory (RAM) rather than just checking the file on the disk?

Thank you for your support.

### ⚠️ Meta-information

```shell
version
Protocol version 48
Exe version 1.1.2.2/Stdio (valve)
ReHLDS version: 3.14.0.857-dev
Build date: 19:52:21 Mar 27 2025 (4002)
Build from: https://github.com/rehlds/ReHLDS/commit/89958d3


] game version (using client.dll cheat)
Protocol version 48
Exe version 1.1.2.0 (valve)
Exe build: 16:05:41 Jun 15 2009 (

meta version
Metamod-r v1.3.0.149, API (5:13)
Metamod-r build: 11:31:17 Apr 23 2024
Metamod-r from: https://github.com/theAsmodai/metamod-r/commit/603a257

meta list
Currently loaded plugins:
      description   stat pend  file                       vers            src  load  unload
 [ 1] AMX Mod X     RUN   -    amxmodx_mm_i386.so         v1.9.0.5293     ini  Start ANY
 [ 2] AgFix         RUN   -    agfix_mm_i386.so           v0.4            ini  ANY   ANY
 [ 3] Reunion       RUN   -    reunion_mm_i386.so         v0.2.0.27       ini  Start Never
 [ 4] Rechecker     RUN   -    rechecker_mm_i386.so       v2.7            ini  Chlvl ANY
 [ 5] FakeMeta      RUN   -    fakemeta_amxx_i386.so      v1.9.0.5294     pl1  ANY   ANY
 [ 6] Ham Sandwich  RUN   -    hamsandwich_amxx_i386.so   v1.9.0.5294     pl1  ANY   ANY
 [ 7] Engine        RUN   -    engine_amxx_i386.so        v1.9.0.5294     pl1  ANY   ANY
 [ 8] Fun           RUN   -    fun_amxx_i386.so           v1.9.0.5294     pl1  ANY   ANY
 [ 9] hackdetector  RUN   -    hackdetector_amxx_i386.so  v0.15.328.lite  pl1  ANY   ANY
 [10] whois         RUN   -    whois_amxx_i386.so         v0.2.270        pl1  ANY   ANY
10 plugins, 10 running

amxx version
AMX Mod X 1.9.0.5293 (http://www.amxmodx.org)
Authors:
        David "BAILOPAN" Anderson, Pavol "PM OnoTo" Marko
        Felix "SniperBeamer" Geyer, Jonny "Got His Gun" Bergstrom
        Lukasz "SidLuke" Wlasinski, Christian "Basic-Master" Hammacher
        Borja "faluco" Ferrer, Scott "DS" Ehlert
Compiled: Oct  7 2021 17:13:53
Built from: https://github.com/alliedmodders/amxmodx/commit/8e8dfa3
Build ID: 5293:8e8dfa3
Core mode: JIT+ASM32

amxx plugins
Currently loaded plugins:
       name                    version     author            file             status
 [  1] Admin Base              1.9.0.5294  AMXX Dev Team     admin.amxx       running
 [  2] Admin Commands          1.9.0.5294  AMXX Dev Team     admincmd.amxx    running
 [  3] Menus Front-End         1.9.0.5294  AMXX Dev Team     menufront.amxx   running
 [  4] Commands Menu           1.9.0.5294  AMXX Dev Team     cmdmenu.amxx     running
 [  5] Players Menu            1.9.0.5294  AMXX Dev Team     plmenu.amxx      running
 [  6] Teleport Menu           1.9.0.5294  AMXX Dev Team     telemenu.amxx    running
 [  7] Maps Menu               1.9.0.5294  AMXX Dev Team     mapsmenu.amxx    running
 [  8] Plugin Menu             1.9.0.5294  AMXX Dev Team     pluginmenu.amxx  running
 [  9] Admin Chat              1.9.0.5294  AMXX Dev Team     adminchat.amxx   running
 [ 10] Pause Plugins           1.9.0.5294  AMXX Dev Team     pausecfg.amxx    running
 [ 11] Stats Configuration     1.9.0.5294  AMXX Dev Team     statscfg.amxx    running
 [ 12] Optimized HPK           1.1         OLO/shadow/Gemin  hpk.amxx         debug
 [ 13] Chat Translator         1.7         abdobiskra        AdminTranslateC  debug
 [ 14] Advanced Arabic Chat H  0.2         abdobiskra        AdvancedArabicC  debug
 [ 15] HL Score Saver          1.0         Flipper_SPb       hlss.amxx        debug
 [ 16] Sudden Death + MSGs     1.3         abdobiskra        SuddenDeathAren  debug
 [ 17] Mirror Damage Optimize  1.4         abdobiskra / Gem  MirrorDamage.am  paused
 [ 18] Arena Winner Control    0.2         Kuma              Arena_Winner_Sp  running
 [ 19] precache                1.1         hjvl & Lev        precache.amxx    running
 [ 20] miniAG AFK              1.6         LetiletiLepestok  afk_arena.amxx   paused
 [ 21] Infinit rpg flight      0.01        LetiLetiLepestok  inf_rpg_flight.  running
 [ 22] Lambda Core             0.6         KORD_12.7         lambda_core.amx  running
 [ 23] CrossServer Chatting    0.2         LetiLetiLepestok  mscxc2.amxx      debug
 [ 24] HackDetector            0.15.lite   Lev @ AGHL.RU De  hackdetector.am  running
 [ 25] SubnetBan               2.2         Lev               subnetban.amxx   debug
25 plugins, 23 running
```

### 📄 Relevant log output

```shell

```

**Comment by that-is-not-great** [comment]
bypassing rechecker is old and easy. only noob cheaters get caught by rechecker. nothing you can do, you rely on client input

**Comment by abdobiskraKGB** [comment]
Another point to note: when the game was running, I replaced the client.dll file with the one containing the cheat, and then I logged into the server, whereupon the cheat was detected and the MD5 code appeared. 
This response, as an addition, might lead us to solve a mystery.
